scripts/anonymization/README.md

```diff
@@ -0,0 +1,322 @@
+| pre_epsilon | 0.5     | Any value between 0 and epsilon (epsilon excluded) | The privacy budget spent on preprocessing of the data. Only necessary for continuous columns |
+| algorithm   | mst     | One of: `mst,aim,dpctgan,patectgan`                | The differential privacy mechanism applied to the data                                       |
+
+```xml
+<anonymization>
+    <table name="tablename">
+        <differential_privacy
+            epsilon="1.5"
+            pre_epsilon="0.75"
+            algorithm="patectgan">
+        </differential_privacy>
+    </table>
+</anonymization>
+```
+
+### Column Information
+
+The columns of a table can be split into four categories:
+
+1. `ignore` - Columns that are ignored by the DP-mechanism. Left untouched
+2. `categorical` - Columns that contain categorical data
+3. `continuous` - Columns of data on a continuous domain. Only numerical columns
+4. `ordinal` - Columns of ordinal meaning. Typically numerical columns with low numers of distinct values
+
+Each type of column requires its own XML-tag with each member of a type given as a separate tag. (See exampels below for clarification)
+
+If no column type information is given:
+
+-   The anonymization process will automatically try to infer the category best suitable for each column.
+-   Automatic anonymization can lead to undesired results or failures in the process.
+-   To counter this, each column can be defined by hand with the suitable tags.
+
+---
+
+**Ignore the ID-column and anonymize automatically:**
+
+```xml
+<anonymization>
+    <table name="item">
+        <differential_privacy>
+            <ignore>
+                <column name="i_id"/>
+            </ignore>
+            <!-- Other types are inferred automatically -->
+        </differential_privacy>
+    </table>
+</anonymization>
+```
+
+The ID-column will be removed from the anonymization process and be added back to the data in its original form before pushing back to the DB! This is useful for primary key columns or columns that contain only NULL values.
+
+---
+
+**Define the type of each column individually:**
+
+```xml
+<anonymization>
+    <table name="item">
+        <differential_privacy>
+            <categorical>
+                <column name="i_name"/>
+                <column name="i_data"/>
+                <column name="i_im_id"/>
+            </categorical>
+            <continuous>
+                <column name="i_price"/>
+            </continuous>
+            <ordinal>
+                <column name="i_id"/>
+            </ordinal>
+        </differential_privacy>
+    </table>
+</anonymization>
+```
+
+<span style="color:red"> **Disclaimer**: As soon as the column types are defined by hand, all columns must be defined. It is not possible to only specify some of the categorical columns and let the algorithms do the rest!</span>
```

Comment:
Question about this: Couldn't you
1) start with inferred/discovered defaults
2) use any additional column details in the config to override those defaults
so that you wouldn't have to either specify no columns or specify all columns?

Reply:
Unfortunately, this is the architecture of the DP-algorithms library that we use. (They do not allow mixed configurations)